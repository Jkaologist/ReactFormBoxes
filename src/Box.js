import React from "react";

/*
  NO STATE HERE
  --------------------------
  PROPS:
  id
  height 
  background color
  width
  remove function
  --------------------------
  Parent is BoxList
  Child of BoxList
*/
function Box ({ id, height="50px", backgroundColor="black", width="50px", remove }) {

  return (
    <div>
      <div className="boxDiv" style={{
        height: height,
        width: width,
        backgroundColor: backgroundColor
      }}>
      </div>
      <button className="removeButton" onClick={remove}>
        Remove Box!
      </button>
    </div>
  )
}

export default Box