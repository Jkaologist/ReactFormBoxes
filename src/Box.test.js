import React from "react";
import { render} from "@testing-library/react";
import Box from "./Box";

// smoke test
it("renders", function () {
  render(<Box />);
});
// snapshot test
it("matches the snapshot", function () {
  const { container } = render(<Box />);
  expect(container).toMatchSnapshot();
});
