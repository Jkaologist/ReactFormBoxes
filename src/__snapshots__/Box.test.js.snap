// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div>
    <div
      class="boxDiv"
      style="height: 50px; width: 50px; background-color: black;"
    />
    <button
      class="removeButton"
    >
      Remove Box!
    </button>
  </div>
</div>
`;
